### DISCUSSION TITLE
Blurry BitmapText after upgrade to v.7

### DISCUSSION THREAD

**Discussion by Clonex** [pixijs/pixijs#9560] [Q&A] [answered]
I've been working on upgrading our codebase to PixiJS 7. 
But I've been experiencing alot of issues with our bitmap fonts getting blurry.
Im not really sure what is causing our issue, as it havent really been a issue with previous versions of PIXI.

Before upgrade:
![image](https://github.com/pixijs/pixijs/assets/408229/7abc08ec-1388-47ee-8461-0059f70c5fa8)

After upgrade:
![image](https://github.com/pixijs/pixijs/assets/408229/19d2f3cf-aaee-49ed-88fc-7c0875f88936)

The first area here uses normal `scale = 1`, and the second area is `scale = 2`

The fonts are loaded as a part of the preloading step, they are just loaded using the `Assets` class, and we only progress on with loading the rest after thoose promises have been resolved..
```javascript
await Assets.load('font/hppixelshd.fnt')
```
To add the text in the game-class I just use the `BitmapText`-class..
```javascript
const temp = new PIXI.BitmapText('Temp', { fontName: 'Hppixels10' });
```

Anyone have any ideas what could be causing the issue?

**Comment by KreNtal** [comment]
Have you tried increasing the fontSize instead of scaling it?

**Comment by Clonex** [comment, marked as answer]
I figured out the issue.
I was setting setting `ROUND_PIXELS`, and `scaleMode` too "late" in the proccess.
```javascript
settings.ROUND_PIXELS = true;
BitmapFont.defaultOptions.scaleMode = SCALE_MODES.NEAREST;
```
I was setting it after all the class attributes had been made. Didn't realize that its saved on the PIXI objects when initializing them.
